File: testProlog/JavaTest/ObStackTest.java
/*
Garrett Addington
This file  is suppose to be in the same location with the other soruce files when building
Testing pop......
First we pushed 10 into the stack.
10 is popped from the stack.

Testing peep......
We push 10 into the stack
We peep on the stack: 10

Testing push........
Current stack size is: 0
12 is pushed to the stack and stack size increased to: 1
*/

package iProlog;

import static org.junit.Assert.*;

import org.junit.Before;
import org.junit.Test;

public class ObStackTest {

	public static ObStack o = new ObStack();

	@Test
	public void testPush() {
		System.out.println("Testing push........");
		if (o.isEmpty()) {
			System.out.println("Initially the stack size is: "+ (o.size() -1));
		}else {
			System.out.println("Initially the stack size is: " +(o.size() -1));
		}
		o.push(12);
		System.out.println(o.get(o.size()-1)+" is pushed to the stack and stack size increased to: "+(o.size() -1));
	}

	@Test
	public void testPop() {
		System.out.println("Testing pop......");
		System.out.println("First we pushed 10 into the stack.");
		o.push(10);
		Object j = o.pop();
		System.out.println(j.toString()+" is popped from the stack.\n");
	}
	@Test
	public void testPeek() {
		System.out.println("Testing peep......");
		System.out.println("We push 10 into the stack");
		o.push(10);
		Object j = o.peek();
		System.out.println("We peep on the stack: "+ j + "\n");
	}
}


/*
Testing pop......
First we pushed 10 into the stack.
10 is popped from the stack.

Testing peep......
We push 10 into the stack
We peep on the stack: 10

Testing push........
Current stack size is: 0
12 is pushed to the stack and stack size increased to: 1

*/

File: testProlog/CppTest/ObStackTest/ObStackTest.cpp
/*
Garrett Addington

Output:
Testing pop......
First we pushed 10 into the stack.
10 is popped from the stack.

Testing peep......
First we pushed 10 into the stack.
We peep on the stack: 10

Testing push.......
Initially the stack size is: 0
12 is pushed to the stack and  stack size increased to:  1

Exact same as the java file.

*/

#include <iostream>
#include "ObStack.h"

using namespace std;

void testPush() {
    ObStack<int> *t = new ObStack<int>;
    cout << "Testing push......." << endl;
    if (t->isEmpty()) {
        cout << "Initially the stack size is: " << t->Obsize() << endl;
    }
    else {
        cout << "Current stack size is: " << t->Obsize() << endl;
    }
    t->push(12);
    cout << t->peek() << " is pushed to the stack and  stack size increased to:  "<< t->Obsize() << endl << endl;
}

void testPop() {
    ObStack<int> *t = new ObStack<int>;
    cout << "Testing pop......" << endl;
    cout << "First we pushed 10 into the stack." << endl;
    t->push(10);
    int popped = t->pop();
    cout << popped << " is popped from the stack." << endl << endl;
}

void testPeek() {
    ObStack<int> *t = new ObStack<int>;
    cout << "Testing peep......" << endl;
    cout << "First we pushed 10 into the stack." << endl;
    t->push(10);
    int peeped = t->peek();
    cout << "We peep on the stack: " << peeped << endl << endl;
}


int main() {
    testPop();
    testPeek();
    testPush();
    return 0;
}
/*Testing pop......
First we pushed 10 into the stack.
10 is popped from the stack.

Testing peep......
First we pushed 10 into the stack.
We peep on the stack: 10

Testing push.......
Initially the stack size is: 0
12 is pushed to the stack and  stack size increased to:  1
 */
